.github/workflows/foundry-tests.yml

```diff
@@ -0,0 +1,47 @@
+name: Tests in foundry
+
+on:
+  workflow_call:
+    secrets:
+      HARDHAT_FORKING_URL:
+        required: true
+
+jobs:
+  foundry-tests:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          submodules: recursive
+
+      - name: Setup node
+        # from tag: v3.2.0
+        uses: actions/setup-node@17f8bd926464a1afa4c6a11669539e9c1ba77048
+        with:
+          node-version: 16.15.0
+
+      - name: Set up node_modules cache
+        # from tag: v2.1.7
+        uses: actions/cache@937d24475381cd9c75ae6db12cb4e79714b926ed
+        with:
+          path: "**/node_modules"
+          key: ${{ runner.os }}-modules-${{ hashFiles('**/package-lock.json') }}
+
+      - name: Install
+        run: npm install
+
+      - name: Install Foundry
+        uses: foundry-rs/foundry-toolchain@v1
+        with:
+          version: nightly
```

Comment:
Should we really use `nightly` build?

Reply:
from  https://github.com/foundry-rs/foundry-toolchain
```
Version to install, e.g. nightly or 1.0.0. Note: Foundry only has nightly builds for the time being.
```

Reply:
I see version 1.0.7: https://github.com/foundry-rs/foundry-toolchain/releases/tag/v1.0.7